techSpec.js

```diff
@@ -0,0 +1,209 @@
+        Later can be reviewed and DRY-ed from then on. 
+        
+        +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+        All 'database connector' functions begin by opening connection to db, and end with closing the connection: 
+
+        let db = new sqlite3.Database(<database path >)  
+        ...
+        db.close()
+        
+        3 - 5. 
+        Each request type requires a different sqlite3 method & SQL query: 
+        
+            GET ALL - should return a summary object
+                    db.each SELECT * FROM table 
+                    ( db.each iterates through result set allowing us to process & return an array of summary objects )
+                    (Alternatively we could have a detail table and a summary table, but would be more complex to update two tables in a POST/PUT request) 
+            
+            GET BY ID - returns a single row 
+                    db.get SELECT * FROM table WHERE id = ? (id passed as argument)
+                    ( db.get deals with a single row result set)
+                    callback > returns the object 
+
+            DEL BY ID - returns no content
+                    db.run DELETE FROM table WHERE id = ? 
+                    
+            PUT BY ID - return updated object
+                    Use db.serialise() wrapper to run: (ensures synchronous execution)
+                        db.run UPDATE table SET keyToBeUpdated = value WHERE id = ?
+                        db.get SELECT * FROM table WHERE id = ?;              
+
+                    Note: As likely updating lots of values, not just one, we may have to use a Conditional update:  
+                    
+                        https://stackoverflow.com/questions/20255138/sql-update-multiple-records-in-one-query 
+                        
+                        UPDATE config
+                        SET config_value = CASE config_name 
+                                    WHEN 'name1' THEN 'value' 
+                                    WHEN 'name2' THEN 'value2' 
+                                    ELSE config_value
+                                    END
+                        WHERE config_name IN('name1', 'name2');
+                        WHERE config_name IN('name1', 'name2');
+                    
+                    
+            POST -  return updated object
+                    The addCat() method will need to pass a generated id as parameter to the database connector function.  
+                    Use db.serialise() wrapper to run:
+                        db.run INSERT INTO table ([arrayOfKeys]) VALUES ([arrayOfValuesToBeUpdated]); 
+                        db.get SELECT * FROM table WHERE id = ?;  
+        
+
+f. Future Goals
+
+    Host on github or similar, to enable access via browser. 
```

Comment:
Sounds good, although it might be worth making it readonly as there's all sorts of automated programs on the internet that do nasty things to APIs without authentication.